FlowCrypt/App/AppContext.swift

```diff
@@ -57,6 +60,7 @@ class AppContext {
             session: sessionType,
             userAccountService: try SessionService(
                 encryptedStorage: encryptedStorage,
+                passPhraseStorage: InMemoryPassPhraseStorage(encryptedStorage: encryptedStorage),
```

Comment:
naming mismatch - passPhraseStorage unclear if memory, stored, or general, with InMemoryPassPhraseStorage passed in

Also, more generally, it's odd that `InMemoryPassPhraseStorage` needs `encryptedStorage` passed into it. Not clear what for?

Reply:
Wouldn't it be more appropriate to use the `combinedPassPhraseStorage` already defined below?

Reply:
I added `encryptedStorage` in `InMemoryPassPhraseStorage` because ClientConfigurationService required `LocalClientConfiguration` which requires `encryptedStorage`
I couldn't find better way to get `clientConfiguration.passphraseSessionLengthInSeconds` from `InMemoryPassPhraseStorage.getPassPhrases` function.

![image](https://user-images.githubusercontent.com/77344191/193052898-1900a01b-b565-494d-8e4f-c2b38ba42855.png)

Reply:
you mean to pass `ClientConfiguration` to `InMemoryPassPhraseStorage` constructor or `InMemoryPassPhraseStorage.getPassPhrases`?

Reply:
`EmailKeyManagerApi` doesn't get initialized in AppContext, meaning that we can use fresh client configuration when initializing it.

Not so with the pass phrase storage classes.

There is another option, instead of full-fledged `encryptedStorage` you could require `clientConfigurationStorage: ClientConfigurationStorage` interface. Then in AppContext you pass `clientConfigurationStorage: encryptedStorage`. `EncryptedStorage` would of course implement that interface. That would work for me: the intent would be clearer.

Reply:
Sometimes these kinds of issues aren't really with the code or with the implementation, but with properly communicating intent through the code to avoid confusion.

Reply:
@tomholub I think that won't work that way.
If we try that way, swift compiler would throw this error.

![image](https://user-images.githubusercontent.com/77344191/193546251-5b791a57-43ef-4dfa-96c8-113e3c26e814.png)

Reply:
@ioanmo226 what about adding `var clientConfiguration: ClientConfiguration?` property to `CombinedPassPhraseStorage`? It will be set on [AppContextWithUser](https://github.com/FlowCrypt/flowcrypt-ios/blob/feature/issue-1805/FlowCrypt/App/AppContext.swift#L164) initialization like this:
```swift
self.clientConfigurationService = ClientConfigurationService(
    server: enterpriseServer,
    local: LocalClientConfiguration(
        encryptedStorage: encryptedStorage
    )
)

combinedPassPhraseStorage.configuration = clientConfigurationService.configuration
super.init(
    ...
)
```
And then we can pass `passphraseSessionLengthInSeconds` value directly to `getPassphrases` methods.

What do you think?

Reply:
@sosnovsky
That sounds good to me and implemented this way.
Please check

Reply:
@ioanmo226 looks much more clear than before 👍